scripts/Calibration/tofpd/diagnostics.py

```diff
@@ -78,3 +118,110 @@ def plot2d(workspace, tolerance: float=0.001, peakpositions: np.ndarray=DIAMOND,
 
     # return the figure so others can customize it
     return fig, fig.axes
+
+
+def difc_plot2d(calib_new, calib_old=None, instr_ws=None, mask=None, vrange=(-1,1)):
+    """
+    Plots the percent change in DIFC between calib_new and calib_old
+    :param calib_new: New calibration, can be filename, SpecialWorkspace2D, or calibration table (TableWorkspace)
+    :param calib_old: Optional old calibration, can be same as types as calib_new; if not specified, default instr used
+    :param instr_ws: Workspace used for instrument definition, only needed if calib_new and/or calib_old are tables
+    :param mask: MaskWorkspace used to hide detector pixels from plot
+    :param vrange: Tuple of (min,max) used for the plot color bar
+    :return: figure and figure axes
+    """
+
+    ws_new = _get_difc_ws(calib_new, instr_ws)
+    if ws_new is None:
+        raise TypeError("Expected to receive a workspace or filename, got None.")
+
+    ws_old = _get_difc_ws(calib_old, instr_ws)
+    if ws_old is None:
+        # If no second workspace is given, then load default instrument to compare against
+        instr_name = ws_new.getInstrument().getName()
+        empty_instr = LoadEmptyInstrument(InstrumentName=instr_name, OutputWorkspace="__def_{}".format(instr_name))
+        ws_old = CalculateDIFC(InputWorkspace=empty_instr, OutputWorkspace="__difc_{}".format(instr_name))
+
+    delta = ws_new - ws_old
+
+    use_mask = False
+    if mask is not None and mtd.doesExist(str(mask)):
+        use_mask = True
+
+    # Plotting below taken from addie/calibration/CalibrationDiagnostics.py
+    theta_array = []
+    phi_array = []
+    value_array = []
+    masked_theta_array = []
+    masked_phi_array = []
+    info = delta.spectrumInfo()
+    for idx, x in enumerate(info):
+        pos = x.position
+        theta = np.arccos(pos[2] / pos.norm())
+        phi = np.arctan2(pos[1], pos[0])
+        if use_mask and mtd[str(mask)].dataY(idx):
```

Comment:
Could this be done more clearly by creating an array of what to mask then do slicing of the data?